Removed the 'Choose file' button, and added a (x) on Insulet uploads.

creswick opened the pull request:
I believe this patch fixes https://trello.com/c/P0TpowaC
I've added some simple conditionals and styling to hide the 'choose file' button
once an ibf file has been selected in the uploader, as well as adding a 'delete'
icon button next to the file name.  That button removes the file from the list and
causes the choose and upload buttons to return to their default state by virtue of
toggling the 'isBlockModeFileChosen' state flag on the Upload component.

I have *not* tested that the proper file is uploaded to the server, but the current
upload behavior should be the same as that of clicking the 'choose file' button
multiple times.

Here's a screenshot of the change:
![image](https://cloud.githubusercontent.com/assets/257712/11769137/1054f44c-a195-11e5-81b3-3ab23818899e.png)

gniezen commented:
Hi @creswick and thanks for the PR. I ran it by our designer @skrugman and she really likes it! She suggested to change the hover color to Tidepool purple, #627CFF.

Unfortunately we're right in the middle of quite a large refactoring of the uploader code (switching to Redux), so I'm not sure how soon we'll be able to merge the PR and if it'll need some changes after the refactoring. I'll keep you posted.

creswick commented:
Thanks @gniezen!

I saw the comment from @skrugman on Trello, and updated the hover state yesterday (using a less variable for that color, I believe), and updated this PR with that change, so that's ready if/when you can merge this.  I'm also happy to make changes as needed, just let me know.

gniezen commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;hgy0uNpI&#x2F;10-0-ux-2-dev-1-tst-uploader-omnipod-bundle-of-small-tasks"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> &lt;0:ux, 2:dev, 1:tst&gt; [uploader] OmniPod bundle of small tasks</a>

gniezen commented:
LGTM. Merging.
